### ISSUE TITLE
Missing content element for AcceptedAtAction in PUT and PATCH (V5.0.0-rc5)

### ISSUE THREAD

**Issue by knoxi** [domaindrivendev/Swashbuckle.AspNetCore#1470]
When generating a description for a simple CRUD controller, then the content element in PUT and PATCH is missing.

My controller looks like this:
 ```
[Consumes("application/json")]
[Produces("application/json")]
[ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
[ProducesResponseType(typeof(void), StatusCodes.Status404NotFound)]
[ApiController]
[ApiVersion("1.0")]
[Route("api/v{version:apiVersion}/[controller]")]
public class OrderDetailController : ControllerBase
{
    ...

    [HttpPost()]
    [ProducesResponseType(StatusCodes.Status201Created)]
    [SwaggerOperation("CreateOrderDetail")]
    public async Task<ActionResult<OrderDto>> CreateEntity([FromBody]OrderDto dto)
    {
        ...
        return CreatedAtAction(GetByIdName, new { id = createdDto.Id, version = "1.0" }, createdDto);
    }

    [HttpPut("{id}")]
    [ProducesResponseType(StatusCodes.Status202Accepted)]
    [SwaggerOperation("UpdateOrderDetail")]
    public async Task<ActionResult<OrderDto>> UpdateEntity(TKey id, [FromBody] OrderDto dto)
    {
        ...
        return AcceptedAtAction("Id", new { id = createdDto.Id, version = "1.0" }, updatedDto);
    }

    [HttpPatch("{id}")]
    [ProducesResponseType(StatusCodes.Status202Accepted)]
    [SwaggerOperation("PatchOrderDetail")]
    public async Task<ActionResult<OrderDto>> PatchEntity(TKey id, [FromBody] JsonPatchDocument<OrderDto> patchDoc)
    {
        ...
        return AcceptedAtAction("Id", new { id = createdDto.Id, version = "1.0" }, patchedDto);
    }

    ...
}
```

and the generated swagger.json looks as following:
```
{
    "openapi": "3.0.1",
    ...
    "paths": {
        "/api/v1.0/OrderDetail": {
            "post": {
              "tags": [
                "OrderDetail"
              ],
              "summary": "CreateOrderDetail",
              "operationId": "CreateOrderDetail",
              "requestBody": {
                "content": {
                  "application/json": {
                    "schema": {
                      "$ref": "#/components/schemas/OrderDetailDto"
                    }
                  }
                }
              },
              "responses": {
                "400": {
                  "description": "Bad Request",
                  "content": {
                    "application/json": {
                      "schema": {
                        "$ref": "#/components/schemas/ProblemDetails"
                      }
                    }
                  }
                },
                "404": {
                  "description": "Not Found",
                  "content": {
                    "application/json": {
                      "schema": {
                        "$ref": "#/components/schemas/ProblemDetails"
                      }
                    }
                  }
                },
                "201": {
                  "description": "Success",
                  "content": {
                    "application/json": {
                      "schema": {
                        "$ref": "#/components/schemas/OrderDetailDto"
                      }
                    }
                  }
                }
              }
            },
            "get": ...
          },
          "/api/v1.0/OrderDetail/{id}": {
            "put": {
              "tags": [
                "OrderDetail"
              ],
              "summary": "UpdateOrderDetail",
              "operationId": "UpdateOrderDetail",
              "parameters": [
                {
                  "name": "id",
                  "in": "path",
                  "required": true,
                  "schema": {
                    "type": "integer",
                    "format": "int32"
                  }
                }
              ],
              "requestBody": {
                "content": {
                  "application/json": {
                    "schema": {
                      "$ref": "#/components/schemas/OrderDetailDto"
                    }
                  }
                }
              },
              "responses": {
                "400": {
                  "description": "Bad Request",
                  "content": {
                    "application/json": {
                      "schema": {
                        "$ref": "#/components/schemas/ProblemDetails"
                      }
                    }
                  }
                },
                "404": {
                  "description": "Not Found",
                  "content": {
                    "application/json": {
                      "schema": {
                        "$ref": "#/components/schemas/ProblemDetails"
                      }
                    }
                  }
                },
                "202": {
                  "description": "Success"
                }
              }
            },
            "patch": {
              "tags": [
                "OrderDetail"
              ],
              "summary": "PatchOrderDetail",
              "operationId": "PatchOrderDetail",
              "parameters": [
                {
                  "name": "id",
                  "in": "path",
                  "required": true,
                  "schema": {
                    "type": "integer",
                    "format": "int32"
                  }
                }
              ],
              "requestBody": {
                "content": {
                  "application/json": {
                    "schema": {
                      "type": "array",
                      "items": {
                        "$ref": "#/components/schemas/Operation"
                      },
                      "nullable": true
                    }
                  }
                }
              },
              "responses": {
                "400": {
                  "description": "Bad Request",
                  "content": {
                    "application/json": {
                      "schema": {
                        "$ref": "#/components/schemas/ProblemDetails"
                      }
                    }
                  }
                },
                "404": {
                  "description": "Not Found",
                  "content": {
                    "application/json": {
                      "schema": {
                        "$ref": "#/components/schemas/ProblemDetails"
                      }
                    }
                  }
                },
                "202": {
                  "description": "Success"
                }
              }
            },
            "delete": ...,
            "get": ...
        }
    }
}        
```
In the PUT and PATCH response 202, the content element is missing, while in the PATCH response 201 the content is set.

Is this an expected behavior? I have expected a content element when the controller is returning one. Or have I missed some default behavior configuration in the swagger registration?

**Comment by domaindrivendev** [comment]
You're (understandably) misunderstanding the semantics of the `ProducesResponseTypeAttribute`. If you use it but don't explicitly assign a value to it's `Type` property, then you're in fact saying that there's _no data_ returned with that response code. You need to update your annotations as follows:

```csharp
   [HttpPut("{id}")]
   [ProducesResponseType(StatusCodes.Status202Accepted, Type = typeof(OrderDto))]
   [SwaggerOperation("UpdateOrderDetail")]
   public async Task<ActionResult<OrderDto>> UpdateEntity(TKey id, [FromBody] OrderDto dto)
   {
       ...
       return AcceptedAtAction("Id", new { id = createdDto.Id, version = "1.0" }, updatedDto);
   }
```

Now, with that said ... I'm not quite sure why this wouldn't be a problem for the `Post` case. I can't explain that right now. Need to investigate that a bit further.

**Comment by domaindrivendev** [comment]
Which version of ASP.NET Core are you using? It looks like it's behavior in this area has changed with newer versions. Previously, you always needed to provide an explicit `Type` with the `ProducesResponseTypeAttribute` to express a return type other than void (i.e. no content). However, in newer versions it seems to be defaulting to the return DTO from the action signature, _but only for certain error codes_.

**Comment by domaindrivendev** [comment]
Regardless, the workaround is to explicitly provide the `Type` value.

The behavior I described is part of `ApiExplorer`, the API metadata layer that ships with ASP.NET Core, and is heavily depended on by Swashbuckle. Therefore, this is technically an upstream issue with ASP.NET Core itself. If you're unhappy with the workaround I've described, then I would encourage you to create an issue there instead.
